File: exercises/small_problems_101/easy_5/letter_counter_1.rb
def word_sizes(sentence)
  stat = Hash.new(0)
  sentence.split(' ').each do |word|
    word.delete('^a-zA-Z')
    stat[word.size] += 1
  end
  stat
end

puts word_sizes('Four score and seven.') == { 3 => 1, 4 => 1, 5 => 1, 6 => 1 }
puts word_sizes('Hey diddle diddle, the cat and the fiddle!') == { 3 => 5, 6 => 1, 7 => 2 }
puts word_sizes("What's up doc?") == { 6 => 1, 2 => 1, 4 => 1 }
puts word_sizes('') == {}

File: frontend/javascript/exercises/easy_3/letter_counter_1.js
function wordSizes(str) {
  var words = str.split(/\s+/);
  var wordCount = {};
  words.forEach(function (word) {
    var length = word.length.toString();
    if (length === '0') {
      return wordCount
    } else if (Object.keys(wordCount).indexOf(length) >= 0) {
      wordCount[length] += 1;
    } else {
      wordCount[length] = 1;
    }
  });
  return wordCount;
}

console.log(wordSizes('Four score and seven.'));                       // { "3": 1, "4" : "1", "5": 1, "6": 1 }
console.log(wordSizes('Hey diddle diddle, the cat and the fiddle!'));  // { "3": 5, "6": 1, "7": 2 }
console.log(wordSizes('What\'s up doc?'));                             // { "6": 1, "2": 1, "4": 1 }
console.log(wordSizes(''));                                            // {}
console.log(wordSizes('Four   score  and      seven.'));               // { "3": 1, "4" : "1", "5": 1, "6": 1 }